src/coreComponents/mesh/generators/VTKFaceBlockUtilities.cpp

```diff
@@ -0,0 +1,633 @@
+  ArrayOfArrays< localIndex > elem2dToEdges( num2dElements );
+  for( int i = 0; i < num2dElements; ++i )
+  {
+    std::set< int > edges;
+    for( auto const & p: allEdges[i] )
+    {
+      edges.insert( pairSigToEdge.at( pairToPairSig( p ) ) );
+    }
+    elem2dToEdges.resizeArray( i, edges.size() );
+    std::copy( edges.cbegin(), edges.cend(), elem2dToEdges[i].begin() );
+  }
+
+  // Here we compute the `face2dToEdges` mapping.
+  // What matters is that each `face2d` matches the proper `edge`.
+  // But we do not pay attention to the ordering of the `face2d`, it will be in the sorting order.
+  auto const cmp = [&pairToPairSig]( std::pair< int, int > const & p0,
+                                    std::pair< int, int > const & p1 ) -> bool
+  {
+    return pairToPairSig( p0 ) < pairToPairSig( p1 );
+  };
+  std::set< std::pair< int, int >, decltype( cmp ) > uniqueEdges( cmp );
+  for( std::vector< std::pair< int, int > > const & edges: allEdges )
+  {
+    uniqueEdges.insert( edges.cbegin(), edges.cend() );
+  }
+  GEOSX_ERROR_IF_NE_MSG( LvArray::integerConversion< localIndex >( uniqueEdges.size() ), num2dFaces, "Internal error: wrong number of edges in the fracture/fault mesh computation." );
+
+  array1d< localIndex > face2dToEdges;
+  face2dToEdges.reserve( num2dFaces );
+  for( std::pair< int, int > const & p: uniqueEdges )
+  {
+    face2dToEdges.emplace_back( pairToEdge( p, nodeToEdges ) );
+  }
+
+  return { face2dToEdges, elem2dToEdges };
+}
+
+/**
+ * @brief Compute the 2d faces (geometrical edges in 2d) to 2d elems (geometrical faces in 3d) mapping.
+ * @param[in] num2dFaces The number of 2d faces (geometrical edges in 3d).
+ * @param[in] face2dToEdges The 2d faces (geometrical edges in 2d) to edges mapping.
+ * @param[in] elem2dToEdges The 2d elem (geometrical faces in 3d) to edges mapping.
+ * @return The mapping.
+ */
+ArrayOfArrays< localIndex > compute2dFaceToElems2d( localIndex const num2dFaces,
+                                                    arrayView1d< localIndex const > face2dToEdges,
+                                                    ArrayOfArraysView< localIndex const > elem2dToEdges )
+{
+  ArrayOfArrays< localIndex > face2dToElems2d( num2dFaces );
+
+  // First, a mapping inversion
+  std::map< int, std::vector< int > > edgesToElems2d;
+  for( localIndex i = 0; i < elem2dToEdges.size(); ++i )
+  {
+    for( int const & e: elem2dToEdges[i] )
+    {
+      edgesToElems2d[e].push_back( i );
+    }
+  }
+  // Then we fill the 2d face -> 2d elems mappings using the 2d face -> edges and edges -> 2d elems chain.
+  for( int fi = 0; fi < num2dFaces; ++fi )
+  {
+    localIndex const & edge = face2dToEdges[fi];
+    std::vector< int > const & elems2d = edgesToElems2d[edge];
+    face2dToElems2d.resizeArray( fi, elems2d.size() );
+    std::copy( elems2d.begin(), elems2d.end(), face2dToElems2d[fi].begin() );
+  }
+
+  return face2dToElems2d;
+}
+
+void importFracture( string const & faceBlockName,
```

Comment:
Maybe it should be called `importFractureNetwork` coz it can actually import multiple fractures at once can't it?

Reply:
Yes that's right.

Reply:
To be precise, it can import one network of multiple fractures, yes!